stock-managment/src/main/java/org/dhis2/android/rtsm/ui/managestock/ManageStockViewModel.kt

```diff
@@ -245,6 +266,131 @@ class ManageStockViewModel @Inject constructor(
+                                if (ruleEffect.ruleAction() is RuleActionAssign &&
+                                    (
+                                        (ruleEffect.ruleAction() as RuleActionAssign).field()
+                                            == config.value?.stockOnHand
+                                        )
+                                ) {
+                                    val data = ruleEffect.data()
+                                    val isValid: Boolean = isValidStockOnHand(data)
+                                    val stockOnHand = if (isValid) data else it.stockOnHand
+                                    addItem(it, cell.value, stockOnHand, !isValid)
+                                    if (!isValid) {
+                                        errors.value[cell.id!!] = resources
+                                            .getString(R.string.stock_on_hand_exceeded_message)
+                                    } else {
+                                        errors.value.remove(cell.id)
+                                    }
+
+                                    _allTableState.value = _allTableState.value.map { tableModel ->
+                                        tableModel.copy(
+                                            tableRows = updateTableRows(tableModel.tableRows, cell)
+                                        )
+                                    }
+                                }
+                            }
+
+                            _screenState.postValue(
+                                TableScreenState(
+                                    tables = allTableState.value,
+                                    selectNext = selectNext
+                                )
+                            )
+                        }
+                    }
+                )
+            }
+        }
+    }
+
+    private fun updateTableRows(
+        tableRowModels: List<TableRowModel>,
+        cell: TableCell
+    ): List<TableRowModel> {
+        return tableRowModels.map { tableRowModel ->
+            if (tableRowModel.values.values.find { tableCell ->
+                tableCell.id == cell.id
+            } != null
+            ) {
+                tableRowModel.copy(
+                    values = updateTableCells(tableRowModel.values, cell)
+                )
+            } else {
+                tableRowModel
+            }
+        }
+    }
+
+    private fun updateTableCells(
+        tableCells: Map<Int, TableCell>,
+        cell: TableCell
+    ): Map<Int, TableCell> {
+        val stockEntry = getPopulatedEntries().find { it.item.id == cell.id }
+        return tableCells.mapValues { (index, tableCell) ->
+            when (index) {
+                0 -> tableCell.copy(
+                    value = stockEntry?.stockOnHand
+                )
+                else -> tableCell.copy(
+                    value = stockEntry?.qty,
+                    error = errors.value[cell.id]
```

Comment:
I think that here we only need to do something like this (maybe there is cleaner way, I didn't think it so much:
```
error = if (stockEntry?.hasError == true) {
                        resources.getString(R.string.stock_on_hand_exceeded_message)
                    } else {
                        null
```
So I think that only with this, we already have error handling working.
What do you think?